sycl/source/detail/device_global_map_entry.cpp

```diff
@@ -47,13 +47,12 @@ DeviceGlobalUSMMem::getZeroInitEvent(const plugin &Plugin) {
             *MZeroInitEvent, Plugin) == info::event_command_status::complete) {
       Plugin.call<PiApiKind::piEventRelease>(*MZeroInitEvent);
       MZeroInitEvent = {};
-      return std::nullopt;
+      return OwnedPiEvent(Plugin);
```

Comment:
nit: Another option is to change reference to a pointer and default to `nullptr`.

Reply:
Reference seems to be in line with how it's normally passed around, so since we have the plugin available here we may as well keep it to the norm. It also means we could have a variant that gets an event after the fact, though that's just a sidenote.